src/__tests__/actions-test.ts

```diff
@@ -305,71 +296,81 @@ describe("actions", () => {
 
   describe("editStages", () => {
     it("dispatches request and success", async () => {
+      const registrationUrl = "/admin/libraries/registration";
       const dispatch = stub();
       const formData = new (window as any).FormData();
       formData.append("registration_stage", "production");
 
-      const fetchMock = stub().returns(new Promise<any>((update, reject) => {
-        update({ status: 200 });
-      }));
-      fetch = fetchMock;
+      fetchMock
+        .mock(registrationUrl, "done");
 
       await actions.editStages(formData)(dispatch);
-      expect(dispatch.callCount).to.equal(2);
+      let fetchArgs = fetchMock.calls();
+
+      expect(dispatch.callCount).to.equal(3);
 
       expect(dispatch.args[0][0].type).to.equal(`${ActionCreator.EDIT_STAGES}_${ActionCreator.REQUEST}`);
       expect(dispatch.args[1][0].type).to.equal(`${ActionCreator.EDIT_STAGES}_${ActionCreator.SUCCESS}`);
 
-      expect(fetchMock.callCount).to.equal(1);
-      expect(fetchMock.args[0][0]).to.equal("/admin/libraries/registration");
-      expect(fetchMock.args[0][1].method).to.equal("POST");
-      expect(fetchMock.args[0][1].body).to.equal(formData);
+      expect(fetchMock.called()).to.equal(true);
+      expect(fetchArgs[0][0]).to.equal(registrationUrl);
+      expect(fetchArgs[0][1].method).to.equal("POST");
+      expect(fetchArgs[0][1].body).to.equal(formData);
+
+      fetchMock.restore();
     });
   });
 
   describe("validateEmail", () => {
     it("validates an email address", async () => {
+      const emailUrl = "/admin/libraries/email";
       const dispatch = stub();
       const formData = new (window as any).FormData();
       formData.append("uuid", "abc");
 
-      const fetchMock = stub().returns(new Promise<any>((update, reject) => {
-        update({ status: 200 });
-      }));
-      fetch = fetchMock;
+      fetchMock
+        .mock(emailUrl, "done");
 
       await actions.validateEmail(formData)(dispatch);
-      expect(dispatch.callCount).to.equal(2);
+      let fetchArgs = fetchMock.calls();
+
+      expect(dispatch.callCount).to.equal(3);
```

Comment:
@adriana-alter these last three set of tests should be calling `dispatch` three times instead of two. It seems that only the failing cases call the dispatch twice. Does this seem right to you? (So then it's request, success, load).

Reply:
No, it seems that certain types of (not-failing) cases are in fact supposed to call the dispatch twice; depends on what the action entails.  This action doesn't load anything; it's just a post request.  There are several tests in `circulation-web` that conform to this pattern; [an example](https://github.com/NYPL-Simplified/circulation-web/blob/master/src/__tests__/actions-test.ts#L301).

Reply:
For the `editBook` action you linked, the server returns `Response("", 200)`. For this test of `validateEmail`, the server returns a dictionary. Both `postForm` functions call the `dispatch` three times (it calls "load" in the `.text()` if/else branch). I'm not sure why the test you linked is only saying it called it twice. I tested this one and it does get called three times in the app, and what gets loaded the dictionary that the server sends. So this, along with the other updates to the dispatch being called three times seems right to me. (@aslagle ).